crates/oxc_semantic/src/jsdoc/parser/jsdoc.rs

```diff
@@ -1,29 +1,125 @@
 use super::jsdoc_tag::JSDocTag;
 use super::parse::parse_jsdoc;
+use oxc_span::Span;
 use std::cell::OnceCell;
+use std::collections::BTreeMap;
 
 #[derive(Debug, Clone)]
 pub struct JSDoc<'a> {
     raw: &'a str,
     /// Cached+parsed JSDoc comment and tags
-    cached: OnceCell<(String, Vec<JSDocTag<'a>>)>,
+    cached: OnceCell<(String, BTreeMap<Span, JSDocTag<'a>>)>,
```

Comment:
Do we need a `BTreeMap` here? If jsdocs are always parsed from top to bottom, then the `Vec` is already sorted.

Reply:
`BtreeMap` is not very performant, we should try and avoid it.

Reply:
Oh, I see.
My original motivation was just to manage `Span` outside of `JSDocTag`.

How about a `FxHashMap` instead?